base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -0,0 +1,136 @@
+    # or if it is the final base image. If the final base image is scratch, then
+    # this is omitted, because we aren't including scratch in the sbom.
+    for index, image in enumerate(base_images_digests):
+        property_name = "konflux:container:is_builder_image:for_stage"
+        property_value = str(index)
+        if index == len(base_images_digests) - 1 and not is_last_from_scratch:
+            property_name = "konflux:container:is_base_image"
+            property_value = "true"
+
+        parsed_image = parse_image_reference_to_parts(image)
+
+        purl = PackageURL(
+            type="oci",
+            name=parsed_image.name,
+            version=parsed_image.digest,
+            qualifiers={
+                "repository_url": parsed_image.repository_without_tag,
+                "tag": parsed_image.tag,
+            },
+        )
+
+        component = {
+            "type": "container",
+            "name": parsed_image.repository_without_tag,
+            "purl": purl.to_string(),
+            "properties": [{"name": property_name, "value": property_value}],
+        }
+        components.append(component)
+
+    return components
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Updates the sbom file with base images data based on the provided files"
+    )
+    parser.add_argument("--sbom", type=pathlib.Path, help="Path to the sbom file", required=True)
+    parser.add_argument(
+        "--base-images-from-dockerfile",
+        type=pathlib.Path,
+        help="Path to the file containing base images extracted from Dockerfile via grep, sed and awk in the buildah "
+        "task",
+        required=True,
+    )
+    parser.add_argument(
+        "--base-images-digests",
+        type=pathlib.Path,
+        help="Path to the file containing base images digests."
+        " This is taken from the BASE_IMAGES_DIGEST tekton result that was generated from"
+        "the output of 'buildah images'",
+        required=True,
+    )
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+
+    args = parse_args()
+
+    with args.base_images_from_dockerfile.open("r") as f:
+        base_images_from_dockerfile = f.read().splitlines()
+
+    with args.base_images_digests.open("r") as f:
+        base_images_digests = f.read().splitlines()
+
+    is_last_from_scratch = False
+    if base_images_from_dockerfile[-1] == "scratch":
+        is_last_from_scratch = True
+
+    with args.sbom.open("r") as f:
+        sbom = json.load(f)
+
+    base_images_sbom_components = get_base_images_sbom_components(base_images_digests, is_last_from_scratch)
+    if "formulation" in sbom:
+        sbom["formulation"].append({"components": base_images_sbom_components})
+    else:
+        sbom.update({"formulation": [{"components": base_images_sbom_components}]})
+
+    with args.sbom.open("w") as f:
+        json.dump(sbom, f, indent=4)
```

Comment:
Is there a specific reason to indent with 4 spaces? IMO we can use the default json.dump behavior to keep the SBOM file smaller.

(Or even use `json.dumps(separators=(',', ':'))` to make it even smaller)

Reply:
Ack, this seems fine then